src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -68,14 +81,106 @@ public int DefaultBufferSize
         }
 
         /// <summary>
-        /// Determines whether null values of properties are ignored or whether they are written to the JSON.
+        /// Determines whether null values are ignored during serialization and deserialization.
+        /// The default value is false.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if this property is set after serialization or deserialization has occurred.
+        /// </exception>
+        public bool IgnoreNullValues
+        {
+            get
+            {
+                return _ignoreNullValues;
+            }
+            set
+            {
+                VerifyMutable();
+                _ignoreNullValues = value;
+            }
+        }
+
+        /// <summary>
+        /// Determines whether read-only properties are ignored during serialization and deserialization.
+        /// A property is read-only if it contains a public getter but not a public setter.
+        /// The default value is false.
         /// </summary>
-        public bool IgnoreNullPropertyValueOnWrite { get; set; }
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if this property is set after serialization or deserialization has occurred.
+        /// </exception>
+        public bool IgnoreReadOnlyProperties
+        {
+            get
+            {
+                return _ignoreReadOnlyProperties;
+            }
+            set
+            {
+                VerifyMutable();
+                _ignoreReadOnlyProperties = value;
+            }
+        }
 
         /// <summary>
-        /// Determines whether null values in the JSON are ignored or whether they are set on properties.
+        /// Gets or sets the maximum depth allowed when reading or writing JSON, with the default (i.e. 0) indicating a max depth of 64.
```

Comment:
Writing too? How?

Reply:
The serializer will track it (not done yet -- see https://github.com/dotnet/corefx/issues/36510)

Reply:
The writer won't let you write beyond a depth of 1k though. So, will the users of the serializer honor that limit?